### PR TITLE
Add VS integration test pipeline

### PR DESCRIPTION
**Pull request by JoeRobich**
Adds the ability to validate Roslyn's integration tests against branch builds of Visual Studio.

Example run: https://devdiv.visualstudio.com/DevDiv/_build/results?buildId=5468100&view=results (microsoft)

### CODE REVIEW

File: azure-pipelines-integration-dartlab.yml [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
...osoft.VisualStudio.ComponentGroup.Web.CloudTools --add Microsoft.VisualStudio.Component.IntelliTrace.FrontEnd --add Microsoft.VisualStudio.Component.DiagnosticTools --add Microsoft.VisualStudio.Component.EntityFramework --add Microsoft.VisualStudio.Component.LiveUnitTesting --add Microsoft.VisualStudio.Component.Debugger.JustInTime --add Component.Microsoft.VisualStudio.LiveShare.2022 --add Microsoft.VisualStudio.Component.WslDebugging --add Microsoft.VisualStudio.Component.IntelliCode --add Microsoft.VisualStudio.Workload.NetWeb --add Microsoft.VisualStudio.Component.ManagedDesktop.Prerequisites --add Microsoft.ComponentGroup.Blend --add Microsoft.VisualStudio.Component.DotNetModelBuilder --add Microsoft.VisualStudio.Component.FSharp.Desktop --add Microsoft.VisualStudio.Component.PortableLibrary --add Microsoft.VisualStudio.Workload.ManagedDesktop --add Microsoft.Net.Component.4.6.TargetingPack --add Microsoft.VisualStudio.Component.VSSDK --add Microsoft.VisualStudio.ComponentGroup.VisualStudioExtension.Prerequisites --add Microsoft.Component.CodeAnalysis.SDK --add Microsoft.VisualStudio.Workload.VisualStudioExtension --installPath "C:\Test\VisualStudio" --quiet --norestart --wait
```
Comment (RikkiGibson) [inline azure-pipelines-integration-dartlab.yml:29]: well that's a mouthful

Comment (JoeRobich) [reply azure-pipelines-integration-dartlab.yml:29]: an alternative would be to commit a .vsconfig file to our .eng folder.

**Review by RikkiGibson — APPROVED** [review]
LGTM :) Would be interested to know the plan for running this on a trigger in the internal repo etc. Maybe another thing we can throw in the dashboard and monitor.

File: azure-pipelines-integration-dartlab.yml (hunk 1 above)
Comment (dibarbet) [reply azure-pipelines-integration-dartlab.yml:29]: Is there a way to potentially break it up?  Most of this is adding the required workloads, might be useful to split that out from the rest of the parameters so its easy to see them

**Review by dibarbet — APPROVED** [review]

File: azure-pipelines-integration-dartlab.yml (hunk 1 above)
Comment (JoeRobich) [reply azure-pipelines-integration-dartlab.yml]: I tried splitting it out. Let's see how it goes https://devdiv.visualstudio.com/DevDiv/_build/results?buildId=5481071&view=results

**Pull request merged**